DSMAJCEexamples/src/main/java/DSMAJCE.java

```diff
@@ -0,0 +1,119 @@
+    }
+
+    private static boolean tokenization() throws NoSuchPaddingException, NoSuchAlgorithmException, InvalidAlgorithmParameterException, InvalidKeyException, IllegalBlockSizeException, BadPaddingException {
+        String algorithm = String.format("AES_256/FPE/NOPADDING", 256, FPE, NOPADDING);
+        Cipher cipher = Cipher.getInstance(algorithm, provider1);
+        SobjectDescriptor sobjDesc = new SobjectDescriptor().name("<Key Name>");
+        KeyObject keyObj = SdkmsKeyService.getKeyObject(sobjDesc);
+        SdkmsAESKey key = new SdkmsAESKey(keyObj);
+        AlgorithmParameters params = cipher.getParameters();
+        cipher.init(Cipher.ENCRYPT_MODE, key, params);
+        String PLAIN = "5200828282828210";
+        System.out.println("Plain text: " + PLAIN);
+        // Encrypted content
+        byte[] cipherBytes = cipher.doFinal(PLAIN.getBytes());
+        System.out.println("FPE mode | tokenized to " + new String(cipherBytes));
+        // decrypt the same content
+        cipher.init(Cipher.DECRYPT_MODE, key, params);
+        byte[] plainBytes = cipher.doFinal(cipherBytes);
+        System.out.println("FPE mode | detokenized to " + new String(plainBytes));
+        return Arrays.equals(PLAIN.getBytes(), plainBytes);
+    }
+    public static void sign_verify(String algorithm, int keySize, String pssParameterDigest, MGF1ParameterSpec mgf1ParameterSpec) throws UnsupportedEncodingException, InvalidKeyException, SignatureException, NoSuchAlgorithmException, InvalidAlgorithmParameterException {
+        SobjectDescriptor sobjDesc = new SobjectDescriptor().name("<Key Name>");
+        KeyObject keyObj = SdkmsKeyService.getKeyObject(sobjDesc);
+        RSAPublicKeyImpl rsaPublicKey = new RSAPublicKeyImpl(keyObj);
+        RSAPrivateKeyImpl rsaPrivateKey = new RSAPrivateKeyImpl(keyObj);
+        byte[] data = "test".getBytes("UTF8");
+        Signature sig = Signature.getInstance("SHA256withRSA", provider2);
+        if (mgf1ParameterSpec != null || pssParameterDigest != null) {
+            sig.setParameter(new PSSParameterSpec(pssParameterDigest, "MGF1", mgf1ParameterSpec, 32, 1));
+        }
+        sig.initSign(rsaPrivateKey);
+        sig.update(data);
+        byte[] signatureBytes = sig.sign();
+
+        assertNotNull(signatureBytes);
+
+        sig.initVerify(rsaPublicKey);
+        sig.update(ByteBuffer.wrap(data));
+
+        assertNotNull(sig);
+        assertEquals(true, sig.verify(signatureBytes));
+    }
+    public static void main(String[] args) throws Exception {
+        System.setProperty("javax.net.ssl.keyStoreType", "jks");
+        System.setProperty("javax.net.ssl.keyStore", "<pkcs12_file_path>");
+        System.setProperty("javax.net.ssl.keyStorePassword", "<keystore_password>");
+
+        provider = SdkmsJCE.getInstance();
+        Security.addProvider(provider);
+
+        provider1 = SdkmsJCE.initialize(FORTANIX_API_ENDPOINT, FORTANIX_API_KEY);
+        Security.addProvider(provider1);
+
```

Comment:
Comment that provider2 demonstrates certificate authorization, and requires the setProperty lines 100-102 to work properly.